Activity handler API

Lull3rSkat3r opened the pull request:
Fixes #96

parente commented on kernel_gateway/services/activity/__init__.py:
```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Jupyter Development Team.
```
A bit weird to have this in the __init__ that gets ref'ed form the handler in the same package. How about an ActivityManager in keeping with the rest of the jupyter setup?

parente commented on kernel_gateway/services/kernels/handlers.py:
```diff
@@ -51,6 +54,54 @@ def get_json_body(self):
+        msg_metadata = json.loads(msg_list[3].decode('UTF-8'))
+        # If the message coming back is a status message, we need to inspect it
+        if msg_metadata['msg_type'] == 'status':
+            msg_content = json.loads(msg_list[6].decode('UTF-8'))
```
Parsing every message when the activity handler endpoints are not exposed will probably degredate performance. Can we add a quick if-check here and bypass all of this if the activity API is not enabled on startup?

parente commented on kernel_gateway/services/activity/handlers.py:
```diff
@@ -0,0 +1,16 @@
+
+    def get(self, **kwargs):
+        if 'kg_list_kernels' not in self.settings or self.settings['kg_list_kernels'] != True:
+            raise tornado.web.HTTPError(403, 'Forbidden')
```
Did we use 403 for the /api/kernels endpoint when listing is disabled? Or 404?

Lull3rSkat3r replied:
I was referencing [this](https://github.com/jupyter-incubator/kernel_gateway/blob/master/kernel_gateway/services/kernels/handlers.py#L35) line of code, which I believe is the  `/api/kernels` endpoint.

parente commented:
Please update the README too to at least indicate that this API exists.

Lull3rSkat3r commented:
Updated the following:
* Moved and renamed `ActivityStore` to be `ActivityManager` in `manager.py`
* Updated README to list the api and how to enable it
* Added logic to not add activities if the `list_kernels` flag is not set.

parente commented on kernel_gateway/services/activity/handlers.py:
```diff
@@ -0,0 +1,16 @@
+
+class ActivityHandler(TokenAuthorizationMixin, CORSMixin, tornado.web.RequestHandler):
+
+    def get(self, **kwargs):
```
Decorate this with `@json_errors`.

https://github.com/jupyter-incubator/kernel_gateway/blob/master/kernel_gateway/services/kernels/handlers.py#L27

parente commented:
I'm good with merging this after the addition of @json_errors.

parente commented on kernel_gateway/services/kernels/handlers.py:
```diff
@@ -51,6 +54,58 @@ def get_json_body(self):
+    about kernel usage.
+    '''
+    def open(self, kernel_id):
+        if self.settings.get('kg_list_kernels', False) == True:
```
`if self.settings.get('kg_list_kernels')` would be enough here. And in the other uses of this as well.

Lull3rSkat3r commented:
Push changes for the boolean checks, but the @json_errors annotation is throwing 500 errors. @parente is going look into the 500 errors.

parente commented:
I'll clean up the tests and @json_errors usage as part of #99 where I hope to have one final solution.
